src/topology/builder.rs

```diff
@@ -111,6 +111,8 @@ pub fn build_pieces(config: &super::Config) -> Result<(Pieces, Vec<String>), Vec
     for (name, source) in &config.sources {
         let (tx, rx) = mpsc::channel(1000);
 
+        let typetag = source.source_type();
```

Comment:
What made you decide not to get these from `typetag`?

Reply:
Well we have SinkOutter etc that trait objectized really early and not allowing us to call the inhernt method. So I figured this was easier.